src/AmbientSounds/ViewModels/SettingsViewModel.cs

```diff
@@ -87,6 +93,47 @@ public bool ResumeOnLaunchEnabled
             set => _userSettings.Set(UserSettingsConstants.ResumeOnLaunchKey, value);
         }
 
+        /// <summary>
+        /// The id of the current output device.
+        /// </summary>
+        public string CurrentOutputDeviceId
+        {
+            get => _currentOutputDeviceId;
+            set
+            {
+                if(_currentOutputDeviceId != value)
+                {
+                    _currentOutputDeviceId = value;
+                    _userSettings.Set(UserSettingsConstants.OutputAudioDeviceId, value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// The current output device.
+        /// </summary>
+        public AudioDeviceDescriptor CurrentOutputDevice
+        {
+            get
+            {
+                var id = CurrentOutputDeviceId;
+                return OutputDevices.Where(x => x.DeviceId == id).FirstOrDefault();
+            }
+
+            set
+            {
+                if (OutputDevices.Any())
```

Comment:
linq doesn't seem necessary here
```suggestion
                if (OutputDevices.Count > 0)
```

Reply:
But also, what happens if the setter takes in a value that doesn't exist inside the collection? I think we should check if the value's id exists inside the collection before performing the set

Reply:
It's there just to ensure that we don't update the CurrentOutputDeviceId before the device list is updated, since we reload the list every time the settings page is opened. There's only one chance that it gets an Id which doesn't exist inside the list. That is when it calls LoadOutputDevicesAsync() to read the saved Id. I add a check there to ensure it's able to fallback to an empty Id.